src/averageAges.js

```diff
@@ -54,8 +80,25 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
-function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+function calculateAverageAgeDiff(people, onlyWithSon = false) {
+  let totalAgeDiff = 0;
+  let motherCount = 0;
+
+  people.forEach((person) => {
```

Comment:
Use nested array methods like `filter` and `map` instead of two nested forEach loops. Example: `const ageDiffs = people.filter(person => !onlyWithSon || person.sex === 'm').map(person => ({person, mother: people.find(mother => mother.name === person.mother)})).filter(({person, mother}) => mother !== undefined).map(({person, mother}) => person.born - mother.born);`